pyscf/pbc/tdscf/rks.py

```diff
@@ -32,15 +33,16 @@ class CasidaTDDFT(rhf.TDA):
 
 def tddft(mf):
     '''Driver to create TDDFT or CasidaTDDFT object'''
-    if mf._numint.libxc.is_hybrid_xc(mf.xc):
-        return TDDFT(mf)
-    else:
+    kpt = getattr(mf, 'kpt', None)
+    if not mf._numint.libxc.is_hybrid_xc(mf.xc) and gamma_point(kpt):
         return CasidaTDDFT(mf)
+    else:
+        return TDDFT(mf)
 
 dft.rks.RKS.TDA           = lib.class_as_method(rhf.TDA)
 dft.rks.RKS.TDHF          = None
-dft.rks.RKS.TDDFTNoHybrid = lib.class_as_method(TDDFTNoHybrid)
-dft.rks.RKS.CasidaTDDFT   = lib.class_as_method(CasidaTDDFT)
+dft.rks.RKS.TDDFTNoHybrid = tddft
+dft.rks.RKS.CasidaTDDFT   = tddft
```

Comment:
Let CasidaTDDFT do CasidaTDDFT. Do not need to select the correct TDDFT for user in this case

Reply:
Just fixed it. Now they look like
```
dft.rks.RKS.TDDFTNoHybrid = tddft
dft.rks.RKS.CasidaTDDFT   = lib.class_as_method(CasidaTDDFT)
dft.rks.RKS.TDDFT         = tddft
```